File: pet-clinic-scala/src/main/scala/com/github/rskupnik/model/Customer.scala
package com.github.rskupnik.model

import java.util
import javax.persistence._

import com.fasterxml.jackson.annotation.JsonIgnore

import scala.annotation.meta.field
import scala.beans.BeanProperty

// Case class implementation

// BeanProperty needed to generate getters and setters

@Entity
case class Customer(
       @(Id@field)
       @(GeneratedValue@field)
       @BeanProperty
       var id: Long,

       @BeanProperty
       var firstName: String,

       @BeanProperty
       var lastName: String,

       @(JsonIgnore@field)
       @(OneToMany@field)(mappedBy = "owner")
       @BeanProperty
       var pets: java.util.List[Pet]) {

  // Need to specify an empty constructor
  def this() {
    this(0, "", "", new util.ArrayList[Pet]())
  }

  override def toString: String = s"$firstName $lastName"

}

File: pet-clinic-java/src/main/java/com/github/rskupnik/petclinic/model/Customer.java
package com.github.rskupnik.petclinic.model;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import java.util.List;

@Entity
public class Customer {

    @Id
    @GeneratedValue
    private Long id;
    private String firstName, lastName;

    @JsonIgnore
    @OneToMany(mappedBy = "owner")
    private List<Pet> pets;

    protected Customer() {

    }

    public Customer(String firstName, String lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public List<Pet> getPets() {
        return pets;
    }

    public void setPets(List<Pet> pets) {
        this.pets = pets;
    }

    @Override
    public String toString() {
        return firstName+" "+lastName;
    }
}
